### PR TITLE
[ntuple] page-storage level RNTuple merging

### PR DESCRIPTION
**Pull request by mxxo**
WIP merge implementation for RNTuples (somewhat related to the `hadd`-level merging PR https://github.com/root-project/root/pull/6101)

### CODE REVIEW

File: tree/ntuple/v7/src/RPageStorage.cxx [hunk 1]
```diff
@@ -201,3 +211,54 @@ void ROOT::Experimental::Detail::RPageSink::CommitCluster(ROOT::Experimental::NT
...
+   // assume all inputs have the same size header and footer
+   auto szFooter = sourceDesc.GetFooterSize();
+   auto szHeader = sourceDesc.GetHeaderSize();
+   std::cout << "source header size " << szHeader << "\n";
+   std::cout << "source footer size " << szFooter << "\n";
+
+   // fill header to get column ids
+   auto headerBuffer = std::make_unique<unsigned char[]>(szHeader);
+   sourceDesc.SerializeHeader(headerBuffer.get());
+   fDescriptorBuilder.SetFromHeader(headerBuffer.get());
```
Comment (mxxo) [inline tree/ntuple/v7/src/RPageStorage.cxx:240]: Here I make a copy of the first RNTuple source header and use it as the merged RNTuple header. This is a bit hacky, but I think it's OK, as long as we override the RNTuple `uuid` at a later point (not implemented yet).  It is a workaround for not having `RColumn` objects to use with `RPageSink::AddColumn(DescriptorId_t fieldId, const RColumn &column)`

File: tree/ntuple/v7/src/RPageStorageFile.cxx [hunk 2]
```diff
@@ -133,6 +133,22 @@ ROOT::Experimental::Detail::RPageSinkFile::CommitPageImpl(ColumnHandle_t columnH
    return result;
 }
 
+ROOT::Experimental::RClusterDescriptor::RLocator
+ROOT::Experimental::Detail::RPageSinkFile::WriteRawPageImpl(
+   ROOT::Experimental::Detail::RPageStorage::RRawPage page)
+{
+   // todo(max) check if setting packedBytes to zippedBytes breaks blob reads
+   auto offsetData = fWriter->WriteBlob(page.fBuffer.get(),
+      /* zippedBytes */ page.fSize, /* packedBytes */ page.fSize);
```
Comment (mxxo) [inline tree/ntuple/v7/src/RPageStorageFile.cxx:142]: I would like to double-check this behaviour is OK (setting `packedBytes` to a possibly incorrect value). At this point we have no way of finding packedBytes without uncompressing the page. `packedBytes` doesn't seem to be used during page reads but I would like to flag this for special consideration during review.

Comment (jblomer) [reply tree/ntuple/v7/src/RPageStorageFile.cxx:142]: Hm, zipped and unzipped bytes (packed bytes) should be used during reading in order to figure out whether or not the page is compressed. If and only if compression doesn't make the page smaller, than the uncompressed page is stored and `packedBytes == zippedBytes`.

**Comment by hageboeck** [comment]
Converted to draft instead of using the label "work in progress".

**Pull request closed without merging**